### DISCUSSION TITLE
Notifications for events

### DISCUSSION THREAD

**Discussion by ascsanto** [davish/obsidian-full-calendar#366] [Feature Requests]
It would be cool an option that have a windows notification pop up (even with a selectable alarm sound) when the time of a new event comes

EDIT: It would be cool to also have an option (like in google calendar) to select which calendars give you the notifications, for example to have notifications only on the "work" calendar

**Comment by JustinasGaripovas** [comment]
Would like to see this feature on mobile as well, to notify X amount of time before event

**Comment by ghost** [comment]
I would also love this feature. It could be possible with the Reminder plugin, if @davish does not want to implement it natively. I just did a PR for it #463

**Reply by pravin-d** [reply]
@zilmosat Trying to install your branch locally. I'm using `yarn` and `node: v18.17.1`. `yarn build` is throwing a lot of TS compile errors. Mostly with obsidian types. 
Also tried running the code in dev mode `yarn dev`. Code runs however the css is not loaded and hangs when I like on 'Save Event' in the create event modal.
Not sure if it issue with my setup, main code itself or it is specific to your branch.
Can you check and let me know.
Really love the concept of reminders. Just the calendar view feels a little incomplete

**Reply by ghost** [reply]
Which ones do you get? I only see these 2: 
```src/calendars/FullNoteCalendar.ts:98:54 - error TS2339: Property 'date' does not exist on type 'Partial<OFCEvent>'.
  Property 'date' does not exist on type 'Partial<{ title: string; id?: string | undefined; } & { allDay: true; } & { type: "recurring"; daysOfWeek: ("U" | "M" | "T" | "W" | "R" | "F" | "S")[]; startRecur?: string | undefined; endRecur?: string | undefined; }>'.

98     return `- [ ] Reminder for this event (@${fields.date ?? "no date"} ${
                                                        ~~~~
src/calendars/FullNoteCalendar.ts:99:16 - error TS2339: Property 'startTime' does not exist on type 'Partial<OFCEvent>'.
  Property 'startTime' does not exist on type 'Partial<{ title: string; id?: string | undefined; } & { allDay: true; } & { type: "single"; date: string; endDate: string | null; completed?: string | false | null | undefined; }>'.

99         fields.startTime ?? "no time"
                  ~~~~~~~~~
Found 2 errors.
```
Those as you mentioned don't have any effect on the code and it runs, still i've fixed it adding 2 checks before reading those fields. With the css I haven't experienced any issues, do you have any console output for it?

**Reply by pravin-d** [reply]
While doing `yarn build`

```
obsidian-full-calendar git:(reminder_plugin) ✗ yarn build
yarn run v1.22.19
$ tsc -noEmit -skipLibCheck && node esbuild.config.mjs production
src/calendars/DailyNoteCalendar.ts:285:23 - error TS2352: Conversion of type 'import("/home/obsidian-full-calendar/node_modules/obsidian-daily-notes-interface/node_modules/obsidian/obsidian").TFile[]' to type 'import("/home/obsidian-full-calendar/node_modules/obsidian/obsidian").TFile[]' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'import("/home/obsidian-full-calendar/node_modules/obsidian-daily-notes-interface/node_modules/obsidian/obsidian").TFile' is not comparable to type 'import("/home/obsidian-full-calendar/node_modules/obsidian/obsidian").TFile'.
    The types returned by 'vault.createFolder(...)' are incompatible between these types.
      Type 'Promise<TFolder>' is not comparable to type 'Promise<void>'.
        Type 'TFolder' is not comparable to type 'void'.

285         const files = Object.values(notes) as TFile[];
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/calendars/DailyNoteCalendar.ts:300:20 - error TS2352: Conversion of type 'import("/home/obsidian-full-calendar/node_modules/obsidian-daily-notes-interface/node_modules/obsidian/obsidian").TFile' to type 'import("/home/obsidian-full-calendar/node_modules/obsidian/obsidian").TFile' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

300         let file = getDailyNote(m, getAllDailyNotes()) as TFile;
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/calendars/DailyNoteCalendar.ts:302:20 - error TS2352: Conversion of type 'import("/home/obsidian-full-calendar/node_modules/obsidian-daily-notes-interface/node_modules/obsidian/obsidian").TFile' to type 'import("/home/obsidian-full-calendar/node_modules/obsidian/obsidian").TFile' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

302             file = (await createDailyNote(m)) as TFile;
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/calendars/DailyNoteCalendar.ts:378:27 - error TS2352: Conversion of type 'import("/home/obsidian-full-calendar/node_modules/obsidian-daily-notes-interface/node_modules/obsidian/obsidian").TFile' to type 'import("/home/obsidian-full-calendar/node_modules/obsidian/obsidian").TFile' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

378             let newFile = getDailyNote(m, getAllDailyNotes()) as TFile;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/calendars/DailyNoteCalendar.ts:380:27 - error TS2352: Conversion of type 'import("/home/obsidian-full-calendar/node_modules/obsidian-daily-notes-interface/node_modules/obsidian/obsidian").TFile' to type 'import("/home/obsidian-full-calendar/node_modules/obsidian/obsidian").TFile' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.

380                 newFile = (await createDailyNote(m)) as TFile;
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 5 errors.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Reply by pravin-d** [reply]
Obsidian type issue ?! slightly confused here

**Reply by ghost** [reply]
That is very strange and should not ocurr, because I did not modify the `DailyNoteCalendar.ts` file. It's an error in that typescript file. Try testing it in a new vault and with a clean installation. I will also continue testing to try to find why the error appears. Here is my output with `npm run build`: 
![Screenshot from 2023-11-01 15-21-32](https://github.com/davish/obsidian-full-calendar/assets/101370810/1341d559-4659-47f6-bf4d-5bd4745a230d)

**Comment by stoweboyd** [comment]
You can implement mannually if using the daily calendar integration. Just requires adding a reminder and making the event a task, like 
`- [ ] zba [startTime:: 19:00]  [endTime:: 21:00] @{2023-09-19}@@{07:00 PM}` 
which is a lot of work. And if you reschedule via reminders, you have to manually reset the full calendar front matter.

**Reply by ghost** [reply]
Yes, that's why I implemented it and created a Pull Request. In my implementation it is added automatically and it is updated when rescheduling it in the calendar. Let's hope that @davish merges it or has a better idea.

**Reply by raven2cz** [reply]
Any progress with PR? I need notifications support for linux.

**Comment by YouFoundJK** [comment]
@ascsanto @JustinasGaripovas and anyone else interested - please check my fork out [here](https://github.com/YouFoundJK/plugin-full-calendar). It has a beta reminder and status bar implemented for Desktop Obsidian (Making it work in mobile is beyond my control - Obsidian doesnt has such an API / way to do it).
